lib/components/user/new-account-wizard.js

```diff
@@ -1,66 +1,76 @@
-import React from 'react'
+import React, { Component } from 'react'
 
 import SequentialPaneDisplay from './sequential-pane-display'
 
 /**
  * This component is the new account wizard.
  */
-const NewAccountWizard = props => {
-  // The props include Formik props that provide access to the current user data (stored in props.values)
-  // and to its own blur/change/submit event handlers that automate the state.
-  // We forward the props to each pane so that their individual controls
-  // can be wired to be managed by Formik.
-  const { panes, values: userData } = props
+class NewAccountWizard extends Component {
+  _handleCreateNewUser = async () => {
+    const {
+      onCreate, // provided by UserAccountScreen
+      setFieldValue, // provided by Formik
+      values: userData // provided by Formik
+    } = this.props
 
-  const {
-    hasConsentedToTerms,
-    notificationChannel = 'email'
-  } = userData
+    const newId = await onCreate(userData)
 
-  const paneSequence = {
-    terms: {
-      disableNext: !hasConsentedToTerms,
-      nextId: 'notifications',
-      pane: panes.terms,
-      props,
-      title: 'Create a new account'
-    },
-    notifications: {
-      nextId: notificationChannel === 'sms' ? 'verifyPhone' : 'places',
-      pane: panes.notifications,
-      prevId: 'terms',
-      props,
-      title: 'Notification preferences'
-    },
-    verifyPhone: {
-      disableNext: true, // TODO: implement verification.
-      nextId: 'places',
-      pane: panes.verifyPhone,
-      prevId: 'notifications',
-      props,
-      title: 'Verify your phone'
-    },
-    places: {
-      nextId: 'finish',
-      pane: panes.locations,
-      prevId: 'notifications',
-      props,
-      title: 'Add your locations'
-    },
-    finish: {
-      pane: panes.finish,
-      prevId: 'places',
-      props,
-      title: 'Account setup complete!'
-    }
+    // After user is initially persisted and reloaded to the redux state,
+    // update the 'id' in the Formik state.
+    setFieldValue('id', newId)
```

Comment:
This function is probably unnecessary and is probably better handled in the application state. It appears that it's purpose is to update a hidden field to associate further saves of the user with it's ID once it is created. I'm not sure how this occurs, but wouldn't Formik already get updated with the user's ID via the `initialValues` prop that is sent the `loggedInUser` in the `user-account-screen#render` method?

Reply:
I thought Formik would get updated too, however it is not the case as seen with https://github.com/opentripplanner/otp-react-redux/pull/224#discussion_r510429924.

Reply:
Seems like this is what should be used to overcome this: https://formik.org/docs/api/withFormik#mappropstovalues-props-props--values

Reply:
Updated (a bit differently) in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/3c4802751a090c2ad4e95a887b7497476a12ba22.